File: web/projects/portal/src/app/common/dnd/table-remove-column-info.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../viewsheet-client/index";
import { ChangeColumnInfo } from "./change-column-info";
import { AssetEntry } from "../../../../../shared/data/asset-entry";

/**
 * Event for dnd.
 */
export class TableRemoveColumnInfo extends ChangeColumnInfo {
   /**
    * Creates a new instance of <tt>TableRemoveColumnInfo</tt>.
    */
   constructor(type: string, index: number, assembly: string,
      objectType: string = "vstable")
   {
      super(assembly, objectType);
      this.type = type;
      this.index = index;
   }

   public type: string;
   private index: number;
}

File: core/src/main/java/inetsoft/web/binding/dnd/TableRemoveColumnInfo.java
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.binding.dnd;

public class TableRemoveColumnInfo extends ChangeColumnInfo {
   public void setType(String rtype) {
      this.type = rtype;
   }

   public String getType() {
      return type;
   }

   public void setIndex(int idx) {
      this.index = idx;
   }

   public int getIndex() {
      return index;
   }
   
   public String toAttribute() {
      String attribute = "";

      if(index != 0) {
         attribute += "_index:" + index;
      }

      return "_type:" + type + attribute;
   }

   private String type;
   private int index;
}